src/v2/Apps/Order/Routes/Shipping/index.tsx

```diff
@@ -146,211 +141,180 @@ export class ShippingRoute extends Component<ShippingProps, ShippingState> {
 
-  getOrderArtwork = () => this.props.order.lineItems?.edges?.[0]?.node?.artwork
+  const getOrderArtwork = () => props.order.lineItems?.edges?.[0]?.node?.artwork
 
-  isCreateNewAddress = () => this.state.selectedAddressID === NEW_ADDRESS
+  const isCreateNewAddress = () => selectedAddressID === NEW_ADDRESS
 
-  isArtaShipping = () => {
-    const addresses = this.getAddressList()
-    const artaShippingEnabled = !!this.getOrderArtwork()?.artaShippingEnabled
+  const isArtaShipping = () => {
+    const addresses = getAddressList()
+    const artaShippingEnabled = !!getOrderArtwork()?.artaShippingEnabled
 
-    const shippingCountry = this.isCreateNewAddress()
-      ? this.state.address.country
+    const shippingCountry = isCreateNewAddress()
+      ? address.country
       : addresses &&
         addresses.find(
-          address => address?.node?.internalID == this.state.selectedAddressID
+          address => address?.node?.internalID == selectedAddressID
         )?.node?.country
 
     return (
-      this.state.shippingOption === "SHIP" &&
+      shippingOption === "SHIP" &&
       artaShippingEnabled &&
       shippingCountry === "US"
     )
   }
 
-  onContinueButtonPressed = async () => {
-    if (this.isArtaShipping() && !!this.state.shippingQuoteId) {
-      this.selectShippingQuote()
+  useEffect(() => {
+    if (
+      isArtaShipping() &&
+      selectedAddressID &&
+      !shippingQuotes &&
+      !shippingQuoteId
+    ) {
+      selectShipping()
+    }
+  }, [shippingOption, shippingQuotes, shippingQuoteId])
+
+  const onContinueButtonPressed = async () => {
+    if (isArtaShipping() && !!shippingQuoteId) {
+      selectShippingQuote()
     } else {
-      this.selectShipping()
+      selectShipping()
     }
   }
 
-  selectShipping = async () => {
-    const {
-      address,
-      shippingOption,
-      phoneNumber,
-      selectedAddressID,
-    } = this.state
-
+  const selectShipping = async () => {
     if (shippingOption === "SHIP") {
-      if (this.isCreateNewAddress()) {
+      if (isCreateNewAddress()) {
         // validate when order is not pickup and the address is new
         const { errors, hasErrors } = validateAddress(address)
         const { error, hasError } = validatePhoneNumber(phoneNumber)
         if (hasErrors && hasError) {
-          this.setState({
-            addressErrors: errors!,
-            addressTouched: this.touchedAddress,
-            phoneNumberError: error!,
-            phoneNumberTouched: true,
-          })
+          setAddressErrors(errors!)
+          setAddressTouched(touchedAddress())
+          setPhoneNumberError(error!)
+          setPhoneNumberTouched(true)
           return
         } else if (hasErrors) {
-          this.setState({
-            addressErrors: errors!,
-            addressTouched: this.touchedAddress,
-          })
+          setAddressErrors(errors!)
+          setAddressTouched(touchedAddress())
```

Comment:
This seems like another good clue that there are some properties that are related & could live within the same state object.